scripts/gameBoard.js

```diff
@@ -1,90 +1,89 @@
-      drawRooms(x,y);
-      (board.find(square => {
-        if(square.xCoordinate === x && square.yCoordinate === y) return true;
-      })).roomNumber = roomNumber;
-    });
-  });
-  console.log(board);
-
-  ctx.restore();
-}
+  createGameBoard() {
+    this.draw.gameBoardFrame(gameBoardWidth,gameBoardHeight,squareSize);
+    for (var i=1; i<=gameBoardWidth;i++) {
+      for (var j=1; j<=gameBoardHeight;j++) {
+        this.draw.gameSquare(i,j);
+        //TODO create method to add info to the board object
+        this.board.push({
+          xCoordinate: i,
+          yCoordinate: j,
+          roomNumber: 0,
+          activeChip: null,
+          bottomChip: null
+        });
+      }
+    }
 
-function drawGameSquare(xPosition,yPosition,) {
-  ctx.fillStyle = squareBorderColour;
-  ctx.fillRect((xPosition-1)*squareSize,(yPosition-1)*squareSize,squareSize,squareSize);
-  ctx.fillStyle = squareColour;
-  ctx.fillRect((xPosition-1)*squareSize+squareBorderSize,(yPosition-1)*squareSize+squareBorderSize,squareSize-2*squareBorderSize,squareSize-2*squareBorderSize);
-}
+    this.rooms.forEach(function(el) {
+      el.forEach(function(element) {
+        var x = element[0], y = element[1], roomNumber = element[2];
+        this.draw.rooms(x,y);
+        (this.getBoard().find(square => {
+          if(square.xCoordinate === x && square.yCoordinate === y) return true;
+        })).roomNumber = roomNumber;
+      }, this);
+    }, this);
+    console.log(this.board);
+  }
 
-function drawGameBoardFrame(width, height, fieldSize) {
-  ctx.lineWidth=gameBoardFrameSize;
-  ctx.strokeRect(gameBoardFrameSize/2,gameBoardFrameSize/2,width*fieldSize+gameBoardFrameSize,height*fieldSize+gameBoardFrameSize);
-  ctx.translate(gameBoardFrameSize,gameBoardFrameSize);
-}
+  getBoard(){
+    return this.board;
+  }
 
-function drawRooms(xPosition,yPosition) {
-  ctx.fillStyle = squareRoomBorderColour;
-  ctx.fillRect((xPosition-1)*squareSize,(yPosition-1)*squareSize,squareSize,squareSize);
-  ctx.fillStyle = squareRoomColour;
-  ctx.fillRect((xPosition-1)*squareSize+squareBorderSize,(yPosition-1)*squareSize+squareBorderSize,squareSize-2*squareBorderSize,squareSize-2*squareBorderSize);
-}
+  handleRandomClickedChip(x,y, player) {
+    var value = Math.floor(Math.random() * Math.floor(6)+1);
+    
 
-function reloadPage() {
-  window.location.reload(false);
-}
+    var boardSquare = this.board.find(square => {
+      if(square.xCoordinate === x && square.yCoordinate === y) return true;
+    });
 
-function getClickCoordinates(xClick,yClick) {
-  var boarder = gameBoardFrameSize*2;
-  var xLimit = (boarder+(squareSize*gameBoardWidth)-2);
-  var yLimit = (boarder+(squareSize*gameBoardHeight)-2);
+    if(boardSquare.bottomChip === null){
+      if(boardSquare.activeChip != null) {
+        boardSquare.bottomChip = boardSquare.activeChip;
+        this.draw.bottomChip(x,y, boardSquare.bottomChip.colour, 3);
+      }
```

Comment:
Here we could send the **chip** object, since the first 3 params are already in the **chip**, plus the offset and handle that the same way it is already done in **draw.chip**.